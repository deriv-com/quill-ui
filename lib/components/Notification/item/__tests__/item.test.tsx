import { render, screen } from "@testing-library/react";
import NotificationItem from "..";

describe("NotificationBanner", () => {
    it("should display correct content", () => {
        render(<NotificationItem message="Test message" />);

        expect(screen.getByText(/Test message/)).toBeInTheDocument();
    });
    it("should have correct title", () => {
        const { container } = render(<NotificationItem title="Test title" />);

        expect(container).toMatchSnapshot();
    });
    it("should apply correct classname", () => {
        const { container } = render(<NotificationItem className="test" />);

        expect(container).toMatchSnapshot();
    });
    it("should render correctly for mobile - without close button", () => {
        const { container } = render(<NotificationItem isMobile />);

        expect(container).toMatchSnapshot();
    });
    it("should render correct icon based on the notification type", () => {
        const { container } = render(<NotificationItem type="warning" />);

        expect(container).toMatchSnapshot();
    });
});
