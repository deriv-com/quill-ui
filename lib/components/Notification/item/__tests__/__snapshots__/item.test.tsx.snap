// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBanner should apply correct classname 1`] = `
<div>
  <a
    class="notification notification-item test"
  >
    <div
      class="body"
    >
      <div
        class="icon info"
      >
        <svg
          fill="#1789E1"
          height="24"
          role="img"
          viewBox="0 0 32 32"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8.375c-2.93 0-5.586 1.563-7.07 4.063-1.446 2.539-1.446 5.624 0 8.124A8.13 8.13 0 0 0 16 24.626c2.89 0 5.547-1.523 7.031-4.062 1.446-2.5 1.446-5.586 0-8.125-1.484-2.5-4.14-4.063-7.031-4.063M16 26.5c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5m-1.562-6.875h.937v-2.5h-.937c-.547 0-.938-.39-.938-.937 0-.508.39-.938.938-.938h1.874a.95.95 0 0 1 .938.938v3.437h.313a.95.95 0 0 1 .937.938c0 .546-.43.937-.937.937h-3.125c-.547 0-.938-.39-.938-.937 0-.508.39-.938.938-.938M16 14c-.703 0-1.25-.547-1.25-1.25 0-.664.547-1.25 1.25-1.25.664 0 1.25.586 1.25 1.25 0 .703-.586 1.25-1.25 1.25"
          />
        </svg>
        <div
          class="badge-unread"
        />
      </div>
      <div
        class="content"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--bold__decoration--default quill-typography__color--default title"
        />
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default message"
        />
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="icon mark-as-read"
      >
        <svg
          height="22"
          role="img"
          viewBox="0 0 13 22"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M12.434 6.566c.246.274.246.684 0 .93l-7.22 7.219c-.273.273-.683.273-.929 0L.566 10.996a.6.6 0 0 1 0-.902.6.6 0 0 1 .903 0l3.254 3.254 6.78-6.782a.644.644 0 0 1 .903 0z"
            />
          </g>
          <defs>
            <clippath
              id="2fb4e3453244533b2fbbdb2477bdd605__a"
            >
              <path
                d="M0 0h13v22H0z"
              />
            </clippath>
          </defs>
        </svg>
      </button>
      <button
        class="icon delete"
      >
        <svg
          height="22"
          role="img"
          viewBox="0 0 13 22"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M5.215 5.063c-.082 0-.137.054-.192.109l-.52.766H8.47l-.52-.766a.23.23 0 0 0-.191-.11zm4.84.875h1.914c.355 0 .656.3.656.656 0 .383-.3.656-.656.656h-.328l-.657 8.887c-.082.93-.82 1.613-1.75 1.613H3.738c-.93 0-1.668-.684-1.75-1.613L1.332 7.25h-.3a.63.63 0 0 1-.657-.656c0-.356.273-.657.656-.657h1.887L3.93 4.435c.273-.41.765-.684 1.285-.684h2.543c.52 0 1.012.273 1.285.684zm.273 1.312H2.645l.656 8.805a.44.44 0 0 0 .437.383h5.496c.22 0 .41-.165.438-.383z"
            />
          </g>
          <defs>
            <clippath
              id="7606a6352379dd74458e5153601178cf__a"
            >
              <path
                d="M0 0h13v22H0z"
              />
            </clippath>
          </defs>
        </svg>
      </button>
    </div>
  </a>
</div>
`;

exports[`NotificationBanner should have correct title 1`] = `
<div>
  <a
    class="notification notification-item"
  >
    <div
      class="body"
    >
      <div
        class="icon info"
      >
        <svg
          fill="#1789E1"
          height="24"
          role="img"
          viewBox="0 0 32 32"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8.375c-2.93 0-5.586 1.563-7.07 4.063-1.446 2.539-1.446 5.624 0 8.124A8.13 8.13 0 0 0 16 24.626c2.89 0 5.547-1.523 7.031-4.062 1.446-2.5 1.446-5.586 0-8.125-1.484-2.5-4.14-4.063-7.031-4.063M16 26.5c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5m-1.562-6.875h.937v-2.5h-.937c-.547 0-.938-.39-.938-.937 0-.508.39-.938.938-.938h1.874a.95.95 0 0 1 .938.938v3.437h.313a.95.95 0 0 1 .937.938c0 .546-.43.937-.937.937h-3.125c-.547 0-.938-.39-.938-.937 0-.508.39-.938.938-.938M16 14c-.703 0-1.25-.547-1.25-1.25 0-.664.547-1.25 1.25-1.25.664 0 1.25.586 1.25 1.25 0 .703-.586 1.25-1.25 1.25"
          />
        </svg>
        <div
          class="badge-unread"
        />
      </div>
      <div
        class="content"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--bold__decoration--default quill-typography__color--default title"
        >
          Test title
        </p>
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default message"
        />
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="icon mark-as-read"
      >
        <svg
          height="22"
          role="img"
          viewBox="0 0 13 22"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M12.434 6.566c.246.274.246.684 0 .93l-7.22 7.219c-.273.273-.683.273-.929 0L.566 10.996a.6.6 0 0 1 0-.902.6.6 0 0 1 .903 0l3.254 3.254 6.78-6.782a.644.644 0 0 1 .903 0z"
            />
          </g>
          <defs>
            <clippath
              id="2fb4e3453244533b2fbbdb2477bdd605__a"
            >
              <path
                d="M0 0h13v22H0z"
              />
            </clippath>
          </defs>
        </svg>
      </button>
      <button
        class="icon delete"
      >
        <svg
          height="22"
          role="img"
          viewBox="0 0 13 22"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M5.215 5.063c-.082 0-.137.054-.192.109l-.52.766H8.47l-.52-.766a.23.23 0 0 0-.191-.11zm4.84.875h1.914c.355 0 .656.3.656.656 0 .383-.3.656-.656.656h-.328l-.657 8.887c-.082.93-.82 1.613-1.75 1.613H3.738c-.93 0-1.668-.684-1.75-1.613L1.332 7.25h-.3a.63.63 0 0 1-.657-.656c0-.356.273-.657.656-.657h1.887L3.93 4.435c.273-.41.765-.684 1.285-.684h2.543c.52 0 1.012.273 1.285.684zm.273 1.312H2.645l.656 8.805a.44.44 0 0 0 .437.383h5.496c.22 0 .41-.165.438-.383z"
            />
          </g>
          <defs>
            <clippath
              id="7606a6352379dd74458e5153601178cf__a"
            >
              <path
                d="M0 0h13v22H0z"
              />
            </clippath>
          </defs>
        </svg>
      </button>
    </div>
  </a>
</div>
`;

exports[`NotificationBanner should render correct icon based on the notification type 1`] = `
<div>
  <a
    class="notification notification-item"
  >
    <div
      class="body"
    >
      <div
        class="icon warning"
      >
        <svg
          fill="#E18D00"
          height="24"
          role="img"
          viewBox="0 0 32 32"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.688 9.82 7.953 22.516c-.078.117-.078.195-.078.312 0 .313.234.547.547.547h15.117c.313 0 .586-.234.586-.547 0-.117-.04-.195-.117-.312L16.273 9.82A.29.29 0 0 0 16 9.625c-.156 0-.234.078-.312.195m-1.602-.976A2.25 2.25 0 0 1 16 7.75c.742 0 1.484.43 1.875 1.094l7.734 12.695c.235.39.391.82.391 1.29 0 1.327-1.094 2.421-2.46 2.421H8.421A2.434 2.434 0 0 1 6 22.828q0-.703.352-1.289zm3.164 12.031c0 .703-.586 1.25-1.25 1.25-.703 0-1.25-.547-1.25-1.25 0-.664.547-1.25 1.25-1.25.664 0 1.25.586 1.25 1.25m-.312-7.187v3.75c0 .546-.43.937-.938.937-.547 0-.937-.39-.937-.937v-3.75c0-.508.39-.938.937-.938a.95.95 0 0 1 .938.938"
          />
        </svg>
        <div
          class="badge-unread"
        />
      </div>
      <div
        class="content"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--bold__decoration--default quill-typography__color--default title"
        />
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default message"
        />
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="icon mark-as-read"
      >
        <svg
          height="22"
          role="img"
          viewBox="0 0 13 22"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M12.434 6.566c.246.274.246.684 0 .93l-7.22 7.219c-.273.273-.683.273-.929 0L.566 10.996a.6.6 0 0 1 0-.902.6.6 0 0 1 .903 0l3.254 3.254 6.78-6.782a.644.644 0 0 1 .903 0z"
            />
          </g>
          <defs>
            <clippath
              id="2fb4e3453244533b2fbbdb2477bdd605__a"
            >
              <path
                d="M0 0h13v22H0z"
              />
            </clippath>
          </defs>
        </svg>
      </button>
      <button
        class="icon delete"
      >
        <svg
          height="22"
          role="img"
          viewBox="0 0 13 22"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M5.215 5.063c-.082 0-.137.054-.192.109l-.52.766H8.47l-.52-.766a.23.23 0 0 0-.191-.11zm4.84.875h1.914c.355 0 .656.3.656.656 0 .383-.3.656-.656.656h-.328l-.657 8.887c-.082.93-.82 1.613-1.75 1.613H3.738c-.93 0-1.668-.684-1.75-1.613L1.332 7.25h-.3a.63.63 0 0 1-.657-.656c0-.356.273-.657.656-.657h1.887L3.93 4.435c.273-.41.765-.684 1.285-.684h2.543c.52 0 1.012.273 1.285.684zm.273 1.312H2.645l.656 8.805a.44.44 0 0 0 .437.383h5.496c.22 0 .41-.165.438-.383z"
            />
          </g>
          <defs>
            <clippath
              id="7606a6352379dd74458e5153601178cf__a"
            >
              <path
                d="M0 0h13v22H0z"
              />
            </clippath>
          </defs>
        </svg>
      </button>
    </div>
  </a>
</div>
`;

exports[`NotificationBanner should render correctly for mobile - without close button 1`] = `
<div>
  <a
    class="notification notification-item__mobile"
  >
    <div
      class="body"
    >
      <div
        class="icon info"
      >
        <svg
          fill="#1789E1"
          height="24"
          role="img"
          viewBox="0 0 32 32"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8.375c-2.93 0-5.586 1.563-7.07 4.063-1.446 2.539-1.446 5.624 0 8.124A8.13 8.13 0 0 0 16 24.626c2.89 0 5.547-1.523 7.031-4.062 1.446-2.5 1.446-5.586 0-8.125-1.484-2.5-4.14-4.063-7.031-4.063M16 26.5c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5m-1.562-6.875h.937v-2.5h-.937c-.547 0-.938-.39-.938-.937 0-.508.39-.938.938-.938h1.874a.95.95 0 0 1 .938.938v3.437h.313a.95.95 0 0 1 .937.938c0 .546-.43.937-.937.937h-3.125c-.547 0-.938-.39-.938-.937 0-.508.39-.938.938-.938M16 14c-.703 0-1.25-.547-1.25-1.25 0-.664.547-1.25 1.25-1.25.664 0 1.25.586 1.25 1.25 0 .703-.586 1.25-1.25 1.25"
          />
        </svg>
        <div
          class="badge-unread"
        />
      </div>
      <div
        class="content"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--bold__decoration--default quill-typography__color--default title"
        />
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default message"
        />
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="icon mark-as-read"
      >
        <svg
          height="22"
          role="img"
          viewBox="0 0 13 22"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M12.434 6.566c.246.274.246.684 0 .93l-7.22 7.219c-.273.273-.683.273-.929 0L.566 10.996a.6.6 0 0 1 0-.902.6.6 0 0 1 .903 0l3.254 3.254 6.78-6.782a.644.644 0 0 1 .903 0z"
            />
          </g>
          <defs>
            <clippath
              id="2fb4e3453244533b2fbbdb2477bdd605__a"
            >
              <path
                d="M0 0h13v22H0z"
              />
            </clippath>
          </defs>
        </svg>
      </button>
      <button
        class="icon delete"
      >
        <svg
          height="22"
          role="img"
          viewBox="0 0 13 22"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M5.215 5.063c-.082 0-.137.054-.192.109l-.52.766H8.47l-.52-.766a.23.23 0 0 0-.191-.11zm4.84.875h1.914c.355 0 .656.3.656.656 0 .383-.3.656-.656.656h-.328l-.657 8.887c-.082.93-.82 1.613-1.75 1.613H3.738c-.93 0-1.668-.684-1.75-1.613L1.332 7.25h-.3a.63.63 0 0 1-.657-.656c0-.356.273-.657.656-.657h1.887L3.93 4.435c.273-.41.765-.684 1.285-.684h2.543c.52 0 1.012.273 1.285.684zm.273 1.312H2.645l.656 8.805a.44.44 0 0 0 .437.383h5.496c.22 0 .41-.165.438-.383z"
            />
          </g>
          <defs>
            <clippath
              id="7606a6352379dd74458e5153601178cf__a"
            >
              <path
                d="M0 0h13v22H0z"
              />
            </clippath>
          </defs>
        </svg>
      </button>
    </div>
  </a>
</div>
`;
